src/NearestNeighbors.jl

```diff
@@ -128,7 +128,7 @@ end
 function MMI.predict(m::KNNRegressor, (tree, y, w), X)
     Xmatrix     = MMI.matrix(X, transpose=true) # NOTE: copies the data
     idxs, dists = NN.knn(tree, Xmatrix, m.K)
-    preds       = zeros(length(idxs))
+    preds       = similar(y, length(idxs))
 
```

Comment:
You need to do something different here, depending on whether `y` is a vector (single target) or a *table* (multi-target). The multi-target case would be simple if it weren't for the weights. One way to proceed  is as follows: (i) convert the table `y` to a matrix, using `mat = MMI.matrix(y)` , (ii) carry on more-or-less as before to get `pred` (a matrix), and (iii) return `MMI.table(pred, names=names, prototype=y)` where `names = Tables.schema(y).names`, to manifest the result as a table with the same type as `y`. See the docstrings for `MMI.matrix` and `MMI.table` for details. Here `MMI = MLJModelInterface`. 

If you are already familiar with the Tables.jl API you might also consider a strategy using the iterator `Tables.rows(y)` that is more performant when `y` is a row-based table, but I'm not sure it be any better than my suggestion for other types (and in particular the case `y` is a `Tables.MatrixTable`. The common use cases for table input would be `DataFrame` and wrapped matrix.